### ISSUE TITLE
Official way to override version catalog version values through commandline

### ISSUE THREAD

**Issue by liutikas** [gradle/gradle#20070]
[Gradle version catalogs](https://docs.gradle.org/current/userguide/platforms.html#sub:central-declaration-of-dependencies) are a great way to centralize dependencies across the codebase. One missing bit of functionality that could make this feature even better - an ability to override version catalog version values through commandline arguments.

### Expected Behavior
I would imagine it would look something like:
```
./gradlew build -Pgradle.libs.versions.groovy="3.0.5"
```

### Current Behavior
This currently has to be manually done within 
```
dependencyResolutionManagement {
    versionCatalogs {
        libs {
            def groovyOverride = System.getenv("GROOVY_VERSION")
            if (groovyOverride != null) {
                version('groovy', groovyOverride)
            }
        }
    }
}
```

### Context
Our CI builds against unrelease versions of certain libraries and ability to do this version change in an official way would be great without having to touch every settings.gradle file (root project, buildSrc, etc)

**Comment by liutikas** [comment]
Example of how we do it in our project today:
https://cs.android.com/androidx/platform/frameworks/support/+/androidx-main:settings.gradle;drc=b9258bd7e19994b99a2c896c66159eb5e25d7c13;l=59

**Comment by adammurdoch** [comment]
The way you've implemented this is the official way to solve this use case. You can define the version catalog in the settings script and then have logic that looks for the override, or whatever else it needs to do to calculate the final version.

You could potentially implement this as a settings plugin that applies a convention to the versions in the version catalog, that looks for a system property for each dependency in the catalog. This could be shared on the plugin portal. But I guess this issue is asking for the convention to be baked into Gradle?

**Comment by liutikas** [comment]
Yeah, i was hoping for a convention. An official plugin from Gradle on the plugin portal also sounds good.

**Comment by JiangHongTiao** [comment]
+1
